mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowClient.java

```diff
@@ -748,11 +748,12 @@ public String getModelVersionDownloadUri(String modelName, String version) {
    * @return A local file or directory ({@link java.io.File}) containing model artifacts
    */
   public File downloadModelVersion(String modelName, String version) {
-    String downloadUri = getModelVersionDownloadUri(modelName, version);
-
+    String scheme = "models";
+    String path = modelName + "/" + version;
+    URIBuilder downloadUriBuilder = new URIBuilder().setScheme(scheme).setPath(path);
     CliBasedArtifactRepository repository = new CliBasedArtifactRepository(null, null,
             hostCredsProvider);
-    return repository.downloadArtifactFromUri(downloadUri);
+    return repository.downloadArtifactFromUri(downloadUriBuilder.toString());
```

Comment:
can we comment that this won't work if the model version download uri points to a single file not a directory due to the way many ArtifactRepository's download_artifacts handles empty subpaths?